File: ApiServer/Model/views/ClassModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ApiServer.Model.views
{
    public static class ClassModel
    {
        public class ClassForView
        {
            public int Id { get; set; }
            public string ClassColor { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }
            public int TotalTag { get; set; }
            public string ImportDisplay { get; set; }

            public string Project { get; set; }
        }
        public class ClassForMerge
        {
            public int Id { get; set; }
            public string ClassColor { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }

            public string Project { get; set; }
        }

        public class ClassForAdd
        {
            public Guid ProjectId { get; set; }
            public string ClassColor { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }
        }

        public class ClassForUpdate
        {
            public int Id { get; set; }
            public string ClassColor { get; set; }
            public string Name { get; set; }
            public string Description { get; set; }
        }
    }
}

File: Client/src/app/shared/models/class.model.ts
export interface IClass{
    id: number;
    code: string;
    name: string;
    description: string;
    totalTag: number;
    classColor: string;
}

export class Class implements IClass {
    constructor(
        public id:number = 0, 
        public code: string = null, 
        public name: string = null, 
        public description: string = null, 
        public totalTag: number =0, 
        public classColor:string='#000000'){

    }
}
export class ClassForAdd{
    projectId: string;
    code: string;
    name: string;
    description: string;
    classColor: string;
}

export class ClassForUpdate{
    id: number;
    code: string;
    name: string;
    description: string;
    classColor: string;
}

export class ClassList{
    constructor(
        public id:number, 
        public name: string, 
        public checked: boolean = false, 
        public classColor:string = '#000000'){}
}